host-controller/src/main/java/org/jboss/as/domain/controller/transformers/KernelAPIVersion.java

```diff
@@ -75,6 +75,8 @@ public enum KernelAPIVersion {
     VERSION_22_0(22, 0, 0),
     // WildFly 30.0.0
     VERSION_23_0(23, 0, 0),
+    // WildFly 31.0.0
+    VERSION_24_0(24, 0, 0),
```

Comment:
Do not specify the version here, right now 24 is in CURRENT. You can use `ModelVersion.create(24, 0, 0 )`  or even `ModelVersion.create(24)` without sticking it to the KernelAPIVersion.java